test/Microsoft.Health.Fhir.Tests.E2E/Rest/Audit/AuditTests.cs

```diff
@@ -0,0 +1,375 @@
+                ae => ValidateExecutedAuditEntry(ae, expectedAction, expectedResourceType, expectedUri, expectedStatusCode, correlationId));
+        }
+
+        private async Task ExecuteAndValidate(Func<FhirClient, Task> clientSetup, HttpStatusCode expectedStatusCode)
+        {
+            if (!_fixture.IsUsingInProcTestServer || !_fixture.FhirClient.SecuritySettings.SecurityEnabled)
+            {
+                // This test only works with the in-proc server with customized middleware pipeline and when security is enabled.
+                return;
+            }
+
+            const string url = "Patient/123";
+
+            // Create a new client with no token supplied.
+            var client = new FhirClient(_fixture.CreateHttpClient(), ResourceFormat.Json);
+
+            await clientSetup(client);
+
+            FhirResponse<OperationOutcome> response = null;
+
+            try
+            {
+                await client.ReadAsync<Patient>(url);
+            }
+            catch (FhirException ex)
+            {
+                response = ex.Response;
+            }
+
+            string correlationId = response.Headers.GetValues(RequestIdHeaderName).FirstOrDefault();
+
+            Assert.NotNull(correlationId);
+
+            var expectedUri = new Uri($"http://localhost/{url}");
+
+            Assert.Collection(
+                _auditLogger.GetAuditEntriesByCorrelationId(correlationId),
+                ae => ValidateExecutedAuditEntry(ae, "read", ResourceType.Patient, expectedUri, expectedStatusCode, correlationId));
+        }
+
+        private void ValidateExecutingAuditEntry(AuditEntry auditEntry, string expectedAction, Uri expectedUri, string expectedCorrelationId)
+        {
+            ValidateAuditEntry(auditEntry, AuditAction.Executing, expectedAction, null, expectedUri, null, expectedCorrelationId);
+        }
+
+        private void ValidateExecutedAuditEntry(AuditEntry auditEntry, string expectedAction, ResourceType? expectedResourceType, Uri expectedUri, HttpStatusCode? expectedStatusCode, string expectedCorrelationId)
+        {
+            ValidateAuditEntry(auditEntry, AuditAction.Executed, expectedAction, expectedResourceType, expectedUri, expectedStatusCode, expectedCorrelationId);
+        }
+
+        private void ValidateAuditEntry(AuditEntry auditEntry, AuditAction expectedAuditAction, string expectedAction, ResourceType? expectedResourceType, Uri expectedUri, HttpStatusCode? expectedStatusCode, string expectedCorrelationId)
+        {
+            Assert.NotNull(auditEntry);
+            Assert.Equal(expectedAuditAction, auditEntry.AuditAction);
+            Assert.Equal(expectedAction, auditEntry.Action);
+            Assert.Equal(expectedResourceType?.ToString(), auditEntry.ResourceType);
+            Assert.Equal(expectedUri, auditEntry.RequestUri);
+            Assert.Equal(expectedStatusCode, auditEntry.StatusCode);
+            Assert.Equal(expectedCorrelationId, auditEntry.CorrelationId);
```

Comment:
>); [](start = 72, length = 2)

Could we add a comment about why we're not validating the claims here?  Just so we know it's by design and not a mistake.

Reply:
Added the comment. We can probably do this by extending the IdentityServer but that will be separate work. I will file an issue for that.